src/Generator/EnumGenerator/Cases/BackedCases.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Laminas\Code\Generator\EnumGenerator\Cases;
+
+use function sprintf;
+
+/**
+ * @internal
+ *
+ * @psalm-immutable
+ */
+final class BackedCases
+{
+    private string $type;
+
+    /** @var list<non-empty-string> */
+    private array $cases;
+
+    /**
+     * @param list<non-empty-string> $cases
+     */
+    private function __construct(string $type, array $cases)
+    {
+        $this->type  = $type;
+        $this->cases = $cases;
+    }
+
+    public function getBackedType(): ?string
+    {
+        return $this->type;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getCases(): array
+    {
+        return $this->cases;
+    }
+
+    /**
+     * @param array<string, string|int> $backedCases
```

Comment:
```suggestion
     * @param array<string, int>|array<string, string> $backedCases
```
Cases can't be mixed, Same [here](https://github.com/laminas/laminas-code/pull/109/files#diff-e301a27a5ae98f04f13cad35f3b4f5d41838e886deb3e0db9779fdc60967e84aR19) 

Also I really wonder if we need the second parameter when we know this?

Reply:
I'll add the suggestion, thanks 👍🏼 

As for the second param, I think having it is better than inferring the type from the array values.